### ISSUE TITLE
I training on a work station, an the speed is lower than on my laptop, what can be the reason?

### ISSUE THREAD

**Issue by neufang** [piskvorky/gensim#628]
I use the gensim 0.12.4
The only difference is I install every dependencies of gensim via anaconda on the workstation. while I dont use anaconda for installation on my laptop.
The log on workstation shows that the parallel queue is not used.  
The workstation has  **Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz with 12 cpu, and 128gb memory**.
```
2016-03-18 17:17:53,176 : INFO : collecting all words and their counts
2016-03-18 17:17:53,176 : INFO : PROGRESS: at sentence #0, processed 0 words, keeping 0 word types
2016-03-18 17:17:53,220 : INFO : PROGRESS: at sentence #10000, processed 97245 words, keeping 14819 word types
2016-03-18 17:17:53,235 : INFO : collected 22068 word types from a corpus of 140323 raw words and 14420 sentences
2016-03-18 17:17:53,249 : INFO : min_count=10 retains 1365 unique words (drops 20703)
2016-03-18 17:17:53,249 : INFO : min_count leaves 102608 word corpus (73% of original 140323)
2016-03-18 17:17:53,253 : INFO : deleting the raw counts dictionary of 22068 items
2016-03-18 17:17:53,254 : INFO : sample=0.0001 downsamples 569 most-common words
2016-03-18 17:17:53,254 : INFO : downsampling leaves estimated 36137 word corpus (35.2% of prior 102608)
2016-03-18 17:17:53,254 : INFO : estimated required memory for 1365 words and 200 dimensions: 4231500 bytes
2016-03-18 17:17:53,254 : INFO : constructing a huffman tree from 1365 words
2016-03-18 17:17:53,292 : INFO : built huffman tree with maximum node depth 13
2016-03-18 17:17:53,294 : INFO : resetting layer weights
2016-03-18 17:17:53,319 : INFO : training model with 1 workers on 1365 vocabulary and 200 features, using sg=0 hs=1 sample=0.0001 and negative=15
2016-03-18 17:17:53,319 : INFO : expecting 14420 examples, matching count from corpus used for vocabulary survey
2016-03-18 17:17:54,321 : INFO : PROGRESS: at 13.82% examples, 74537 words/s
2016-03-18 17:17:55,323 : INFO : PROGRESS: at 28.29% examples, 76381 words/s
2016-03-18 17:17:56,327 : INFO : PROGRESS: at 42.07% examples, 75792 words/s
2016-03-18 17:17:57,328 : INFO : PROGRESS: at 55.11% examples, 74526 words/s
2016-03-18 17:17:58,330 : INFO : PROGRESS: at 72.72% examples, 78704 words/s
2016-03-18 17:17:59,330 : INFO : PROGRESS: at 86.59% examples, 78090 words/s
2016-03-18 17:18:00,334 : INFO : PROGRESS: at 99.63% examples, 77007 words/s
2016-03-18 17:18:00,354 : INFO : reached end of input; waiting to finish 3 outstanding jobs
2016-03-18 17:18:00,364 : INFO : training on 2104845 raw words took 7.0s, 76906 trained words/s
```
**It even slower when i set 12 workers**.
```
2016-03-18 17:19:06,749 : INFO : expecting 14420 examples, matching count from corpus used for vocabulary survey
2016-03-18 17:19:07,765 : INFO : PROGRESS: at 3.24% examples, 16718 words/s
2016-03-18 17:19:08,801 : INFO : PROGRESS: at 6.24% examples, 16724 words/s
2016-03-18 17:19:09,845 : INFO : PROGRESS: at 9.85% examples, 17182 words/s
2016-03-18 17:19:10,874 : INFO : PROGRESS: at 13.82% examples, 18182 words/s
2016-03-18 17:19:11,878 : INFO : PROGRESS: at 17.71% examples, 18662 words/s
2016-03-18 17:19:12,929 : INFO : PROGRESS: at 21.13% examples, 18530 words/s
2016-03-18 17:19:13,977 : INFO : PROGRESS: at 24.46% examples, 18305 words/s
2016-03-18 17:19:15,045 : INFO : PROGRESS: at 27.74% examples, 18150 words/s
2016-03-18 17:19:16,057 : INFO : PROGRESS: at 32.22% examples, 18789 words/s
2016-03-18 17:19:17,065 : INFO : PROGRESS: at 36.01% examples, 18921 words/s
2016-03-18 17:19:18,073 : INFO : PROGRESS: at 39.57% examples, 18973 words/s
2016-03-18 17:19:19,077 : INFO : PROGRESS: at 43.32% examples, 19014 words/s
2016-03-18 17:19:20,085 : INFO : PROGRESS: at 47.11% examples, 19146 words/s
2016-03-18 17:19:21,087 : INFO : PROGRESS: at 51.83% examples, 19589 words/s
2016-03-18 17:19:22,088 : INFO : PROGRESS: at 55.80% examples, 19747 words/s
2016-03-18 17:19:23,089 : INFO : PROGRESS: at 59.69% examples, 19813 words/s
2016-03-18 17:19:24,093 : INFO : PROGRESS: at 63.71% examples, 19882 words/s
2016-03-18 17:19:25,105 : INFO : PROGRESS: at 67.13% examples, 19824 words/s
2016-03-18 17:19:26,121 : INFO : PROGRESS: at 70.55% examples, 19728 words/s
2016-03-18 17:19:27,148 : INFO : PROGRESS: at 74.80% examples, 19887 words/s
2016-03-18 17:19:28,189 : INFO : PROGRESS: at 78.09% examples, 19756 words/s
2016-03-18 17:19:29,197 : INFO : PROGRESS: at 81.37% examples, 19672 words/s
2016-03-18 17:19:30,205 : INFO : PROGRESS: at 84.97% examples, 19659 words/s
2016-03-18 17:19:31,233 : INFO : PROGRESS: at 89.46% examples, 19817 words/s
2016-03-18 17:19:32,235 : INFO : PROGRESS: at 93.07% examples, 19819 words/s
2016-03-18 17:19:33,241 : INFO : PROGRESS: at 98.24% examples, 20106 words/s
2016-03-18 17:19:33,254 : INFO : reached end of input; waiting to finish 36 outstanding jobs
2016-03-18 17:19:33,785 : INFO : training on 2104845 raw words took 27.0s, 20062 trained words/s
```
My laptop has **Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz with 4 cpu, and 16gb memory.**
```
2016-03-18 17:23:00,532 : INFO : expecting 14420 sentences, matching count from corpus used for vocabulary survey
2016-03-18 17:23:01,552 : INFO : PROGRESS: at 26.15% examples, 198746 words/s, in_qsize 24, out_qsize 0
2016-03-18 17:23:02,567 : INFO : PROGRESS: at 58.94% examples, 225142 words/s, in_qsize 24, out_qsize 0
2016-03-18 17:23:03,603 : INFO : PROGRESS: at 93.77% examples, 237737 words/s, in_qsize 14, out_qsize 0
2016-03-18 17:23:03,662 : INFO : worker thread finished; awaiting finish of 11 more threads
2016-03-18 17:23:03,662 : INFO : worker thread finished; awaiting finish of 10 more threads
2016-03-18 17:23:03,667 : INFO : worker thread finished; awaiting finish of 9 more threads
2016-03-18 17:23:03,693 : INFO : worker thread finished; awaiting finish of 8 more threads
2016-03-18 17:23:03,695 : INFO : worker thread finished; awaiting finish of 7 more threads
2016-03-18 17:23:03,701 : INFO : worker thread finished; awaiting finish of 6 more threads
2016-03-18 17:23:03,702 : INFO : worker thread finished; awaiting finish of 5 more threads
2016-03-18 17:23:03,702 : INFO : worker thread finished; awaiting finish of 4 more threads
2016-03-18 17:23:03,703 : INFO : worker thread finished; awaiting finish of 3 more threads
2016-03-18 17:23:03,708 : INFO : worker thread finished; awaiting finish of 2 more threads
2016-03-18 17:23:03,711 : INFO : worker thread finished; awaiting finish of 1 more threads
2016-03-18 17:23:03,721 : INFO : worker thread finished; awaiting finish of 0 more threads
2016-03-18 17:23:03,721 : INFO : training on 2104845 raw words (776834 effective words) took 3.2s, 244470 effective words/s
```
I have installed cpython, blas, numpy, scipy

**Comment by gojomo** [comment]
Foremost: are you sure of the gensim version on the workstation? Anaconda sometimes only has an older version. 

Latest released gensim has fixes which help somewhat for very-small text examples. (Yours appear to be on average 10 words, and perhaps just 3-4 words each after `min_count` and `sample` applied.) 

Latest gensim also changes some of the default parameters, so unless you are being explicit in all initialization parameters, if your gensim versions diverge they might not be attempting the same kind of training. (In particular, your workstation output suggests *both* hierarchical-softmax and negative-sampling are being used in parallel, which is slower, more memory-intensive, and usually not necessary.)

**Comment by neufang** [comment]
@gojomo  thank you for your quick answers.
I got rid of anaconda on the work station, ad install gensim using pip, now the speed is way faster.
I  gonna try setting only hierarchical-softmax .

**Comment by gojomo** [comment]
You're welcome! In what I've observed, projects with a lot of training data tend to do negative-sampling only, and often fewer negative examples – eg `hs=0, negative=5` – though I've also sometimes seen 15-20 negative examples used with smaller datasets. 

Closing this issue as there's no bug (or if there is, it's likely something in older gensim already improved). Please prefer to use the project discussion list – https://groups.google.com/forum/#!forum/gensim – for any general questions/gotchas.

**Comment by piskvorky** [comment]
@neufang can you raise this issue with Anaconda?

We've had quite a few issue reports because of their platform. This is worrying as Continuum are pushing Anaconda as "the" platform for package management in Python, but we do not have resources to do support for them.
